py/redrock/external/desi.py

```diff
@@ -322,7 +333,18 @@ def __init__(self, spectrafiles, coadd=True, targetids=None,
                 hdata = None
                 if comm_rank == 0:
                     hdata = hdus[extname].data[rows]
-
+                    # as for flux,
+                    # check for NaN here (should never happen of course)
+                    # we do a sum on waves to identify Inf values as well
+                    isnan=np.isnan(np.sum(hdata,axis=1))
+                    if np.sum(isnan)>0 :
+                        # NaN or Inf values
+                        # We set flux to 0 for those
+                        # and will set ivar to 0 after
+                        # for entries with flux==0
+                        for i in np.where(isnan)[0] :
+                            hdata[i][np.isnan(hdata[i])]=0.
+                            hdata[i][np.isinf(hdata[i])]=0. 
```

Comment:
I think this could be handled more simply with:
```python
bad = np.isnan(hdata) | np.isinf(hdata) | np.isneginf(hdata)
hdata[bad] = 0.0
```

I'm more concerned with the assumption downstream that `flux=0` data was actually bad data and should be set to `ivar=0`.  Real data will very very rarely be exactly 0, but that seems fragile.  Would be better to explicitly make a `badflux` mask like above, and then use that mask when loading the ivar HDU to set `ivar[badmask]=0.`